src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -398,49 +345,47 @@ event::ptr loop_inst::set_output_memory(memory::ptr mem, bool check, size_t idx)
     return ev;
 }
 
-loop_inst::concatenated_memory_mapping::ptr loop_inst::create_concat_memory_map(const input_info& internal_id,
-                                        const cldnn::loop::io_primitive_map& io_prim_map,
-                                        memory::ptr mem_ptr,
-                                        const int64_t trip_count) {
+loop_inst::concatenated_memory_mapping::ptr loop_inst::create_concat_memory_map(const cldnn::loop::io_primitive_map& io_prim_map,
+                                                                                    memory::ptr mem_ptr,
+                                                                                    const int64_t num_iterations) {
+    const auto& external_id = io_prim_map.external_id;
+    const auto& internal_id = io_prim_map.internal_id;
     auto& engine = body_network->get_engine();
     auto& stream = body_network->get_stream();
     auto prim = body_network->get_primitive(internal_id.pid);
-    const int64_t start = io_prim_map.start < 0? trip_count - 1: io_prim_map.start;
+    int64_t start = io_prim_map.start < 0? num_iterations - 1: io_prim_map.start;
+    start = start < 0 ? 0 : start;
```

Comment:
This is the case where iter count is -1 and direction is backward. 
Is it correct to set start as 0? 
Shouldnt we do this setup at inference time when the num_iter is known ?

Reply:
It is redundant code. I changed the code to use io_prim_map instead of start. as you mentioned, it is calculated at the runtime. I missed to remove the code during refactoring. I will remove it. thanks.